audio-worklet/noisegate-test.html

```diff
@@ -0,0 +1,451 @@
+
+        // The threshold is between the aboveThresholdGain and
+        // belowThresholdGain. If the threshold is close to the
+        // belowThresholdGain, then the gate may oscillate between opened and
+        // closed, causing the test to fail. The choice of 0.5 for threshold
+        // is arbitrary and can be any positive value.
+        const linearThreshold = 0.5;
+        const scaledThreshold = linearThreshold * 2;
+        const threshold = NoiseGate.toDecibel(scaledThreshold);
+        const aboveThresholdGain = scaledThreshold * 1.25;
+        const belowThresholdGain = scaledThreshold * 0.75;
+
+        // Attack and release are both 0.
+        let noisegate =
+            new NoiseGate(context, {bufferSize, threshold, timeConstant});
+        let gain = new GainNode(context, {gain: aboveThresholdGain});
+        let merger = new ChannelMergerNode(context, {numberOfInputs: 2});
+        
+        // The delay node offsets the script processor latency.
+        let delay = new DelayNode(
+            context, {delayTime : toTime_(scriptProcessorLatency, sampleRate)});
+        let osc = new OscillatorNode(context);
+
+        osc.connect(gain).connect(noisegate.input);
+        gain.connect(delay);
+        noisegate.output.connect(merger, 0, 0);
+        delay.connect(merger, 0, 1);
+        merger.connect(context.destination);
+
+        gain.gain.setValueAtTime(belowThresholdGain, timeToThreshold);
+
+        // Drop the bass!
+        osc.start();
+
+        context.startRendering().then((buffer) => {
+          let noiseGateOutput = buffer.getChannelData(0);
+          let originalOutput = buffer.getChannelData(1);
+         
+          const closedIndex =
+              toIndex_(timeToThreshold, sampleRate) + scriptProcessorLatency;
+
+          // The output is divided into a latency phase, an unaltered phase, and
+          // a closed phase.
+          let latencyPhase = noiseGateOutput.slice(0, scriptProcessorLatency);
+          let unalteredPhase =
+              noiseGateOutput.slice(scriptProcessorLatency, closedIndex);
+          let unalteredPhaseReference =
+              originalOutput.slice(scriptProcessorLatency, closedIndex);
+
+          // The first samples run through the noise gate will be muted because
+          // of envelope delay.
+          const openingDelay =
+              getFirstIndexBeforeRemainingSamplesMatch_(
+                  unalteredPhase, unalteredPhaseReference);
+          
+          // The observed delay of the envelope is 0.002685546875s, and changes
+          // depending on choice of time constant. Unlike the test above, the
+          // delay cannot be easily predicted since the envelope measures
+          // a sine wave.
+          const delayInSeconds = toTime_(openingDelay, sampleRate);
+          should(delayInSeconds,
+                 'The envelope delay for the very first samples')
+              .beCloseTo(0.0026, {threshold: 0.1});
```

Comment:
No, the variation across sample rates is larger. At a sample rate of 2^18, the delay is  0.00275421142578125s which is a relative error of 0.0255. I will set it at 0.05.